esp32/mods/modpycom.c

```diff
@@ -162,8 +163,41 @@ STATIC mp_obj_t mod_pycom_pulses_get (mp_obj_t gpio, mp_obj_t timeout) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_2(mod_pycom_pulses_get_obj, mod_pycom_pulses_get);
 
+
+
+/*
+ * nvs_setstring/nvs_getstring: support for NVS string storage and, 
+ * thanks to JSON/other serializers, pretty much any datastructure.
+ */
+STATIC mp_obj_t mod_pycom_nvs_setstring (mp_obj_t _key, mp_obj_t _value) {
+    const char *key = mp_obj_str_get_str(_key);
+    const char *value = mp_obj_str_get_str(_value);
+    if (strlen(value) >= 1984) {
```

Comment:
Where does 1984 come from ? As per [documentation](https://docs.espressif.com/projects/esp-idf/en/latest/api-reference/storage/nvs_flash.html) it is 5000.

Reply:
Well, this pull requests is getting pretty old...  
If you look at, say, https://docs.espressif.com/projects/esp-idf/en/v2.1.1/api-reference/storage/nvs_flash.html#keys-and-values

there's a note (in blue) "String and blob values are currently limited to 1984 bytes. For strings, this includes the null terminator."

Reply:
Yes, you are right, only in the newest versions (3.1, 3.2) it can be larger than 1984 bytes.